File: docs/images/0009-Win32OLE/ie.py
import win32com.client
from win32com.server import util
import pythoncom

class WebBrowserEvent:
    def OnDownloadComplete(self,*args,**kwds):
        print "Download Complete"
    def OnQuit(self,*args,**kwds):
        exit

def com_collection_iter(collection):
    return (collection.Item(index)
            for index in range(1, collection.Count+1))

#ie = win32com.client.Dispatch("InternetExplorer.Application")
ie = win32com.client.DispatchWithEvents("InternetExplorer.Application",WebBrowserEvent)
ie.Visible = True
ie.GoHome()
while ie.ReadyState != win32com.client.constants.READYSTATE_COMPLETE:
    pass

ie.Navigate('http://www.ruby-lang.org/')

while ie.ReadyState != win32com.client.constants.READYSTATE_COMPLETE:
    pass

count = 0
for element in ie.Document.all:
    count += 1
print "complete\n %d elements found\n" % (count)

File: docs/images/0009-Win32OLE/ie.js
var ie = WScript.CreateObject("InternetExplorer.Application","ie_");

ie.Visible = 1;
ie.GoHome();
while (ie.ReadyState != 4) {
  WScript.Sleep(1000);
}

ie.Document.onclick = function (){
  WScript.Echo("click!!");
};


ie.Navigate('http://www.ruby-lang.org/');
while (ie.ReadyState != 4) {
  WScript.Sleep(1000);
}

var element,count=0;
for(element in ie.Document.all){
  count += 1
}

WScript.Echo("complete\n" + count + " elements found\n");

function ie_DownloadComplete() {
  WScript.Echo("Download Complete");
}

